main.py

```diff
@@ -0,0 +1,95 @@
+        self.api.login()
+
+    def get_user_id(self, name):
+        """Returns id user."""
+        self.api.searchUsername(name)
+        return self.api.LastJson["user"]["pk"]
+
+    def get_followers(self, user_id):
+        """Returns list of user's followers."""
+        followers = self.api.getTotalFollowers(user_id)
+        return [follower['username'] for follower in followers]
+
+    def get_followings(self, user_id):
+        """Returns list of user's followings."""
+        followings = self.api.getTotalFollowings(user_id)
+        return [following['username'] for following in followings]
+
+    def save_db(self, followers, followings):
+        engine = create_engine(f'sqlite:///users.db')
+        Base.metadata.create_all(engine)
+        Session = sessionmaker(bind=engine)
+        session = Session()
+        now = datetime.datetime.now().strftime('%Y-%m-%d %H:%M')
+
+        for (followings, followers) in itertools.zip_longest(followings, followers, fillvalue=''):
+            text = InstagramUsers(followings=followings, followers=followers, status="new", time=str(now))
+            session.add(text)
+            session.commit()
+
+    def user(self):
+        engine = create_engine(f'sqlite:///users.db')
+        Base.metadata.create_all(engine)
+        Session = sessionmaker(bind=engine)
+        session = Session()
+
+        now = datetime.datetime.now().strftime('%Y-%m-%d %H:%M')
+
+        database_followings = session.query(InstagramUsers.followings).all()
+        database_follower = session.query(InstagramUsers.followers).all()
+
+        followings_list = [y for x in database_followings for y in x]
+        followers_list = [y for x in database_follower for y in x]
+
+        for following in followings:  # це  в бд людей на яких я підписався недавно
+            if following not in followings_list:
+                if len(following) >= 1:
+                    text = InstagramUsers(followings=following, followers='', status="subscribed", time=str(now))
+                    session.add(text)
+                    session.commit()
+
+        for following_l in followings_list:  # це запис в бд людей від яких відписався я недавно
+            if following_l not in followings:
+                if len(following_l) >= 1:
+                    text = InstagramUsers(followings=following_l, followers='', status="unsubscribed", time=str(now))
+                    session.add(text)
+                    session.commit()
+
+        for follower in followers:  # це в бд людей які підписались на мене  недавно
+            if follower not in followers_list:
+                if len(follower) >= 1:
+                    text = InstagramUsers(followings='', followers=follower, status="subscribed", time=str(now))
+                    session.add(text)
+                    session.commit()
+
+        for follower_l in followers_list:  # це в бд людей які відписались від мене  недавно
```

Comment:
These all parts looks awful.

It's better to create separate methods or something like that instead of code duplications.

Reply:
done